apm-agent-plugins/apm-jdbc-plugin/src/main/java/co/elastic/apm/agent/jdbc/helper/ConnectionMetaData.java

```diff
@@ -165,36 +175,136 @@ ConnectionMetaData parse(String connectionUrl, String user) {
+                        if (hostPort.port > 0) {
+                            port = hostPort.port;
+                        }
+                    } else {
+                        // Thin driver host:port:sid syntax:
+                        // myhost:666:instance
+                        // myhost:instance
+                        // thin:myhost:port:instance
+                        if (connectionUrl.startsWith("thin:")) {
+                            connectionUrl = connectionUrl.substring("thin:".length());
+                        }
+
+                        String[] parts = connectionUrl.split(":");
+                        if (parts.length > 0) {
+                            host = parts[0];
+                        }
+                        if (parts.length > 1) {
+                            try {
+                                port = Integer.parseInt(parts[1]);
+                            } catch (NumberFormatException e) {
+                                // apparently not a port...
+                            }
+                        }
                     }
+                }
 
-                    String[] parts = connectionUrl.split(":");
-                    if (parts.length > 0) {
-                        host = parts[0];
+                return new ConnectionMetaData(dbVendor, host, port, user);
+            }
+
+            private HostPort parseAddressList(String connectionUrl) {
+                TreeNode parsedTree = null;
+                Deque<TreeNode> stack = new ArrayDeque<>();
+                StringBuilder currentValueBuffer = null;
+                for (char c : connectionUrl.toLowerCase().toCharArray()) {
+                    switch (c) {
+                        case '(': {
+                            TreeNode treeNode = new TreeNode();
+                            if (stack.isEmpty()) {
+                                parsedTree = treeNode;
+                            } else {
+                                stack.peek().childNodes.add(treeNode);
+                            }
+                            stack.push(treeNode);
+                            currentValueBuffer = treeNode.name;
+                            break;
+                        }
+                        case ')': {
+                            stack.pop();
+                            // continue to do the same as finding `=`
+                        }
+                        case '=': {
+                            if (stack.isEmpty()) {
+                                currentValueBuffer = null;
+                            } else {
+                                currentValueBuffer = stack.peek().value;
+                            }
+                            break;
+                        }
+                        default: {
+                            if (currentValueBuffer == null) {
+                                logger.warn("Failed to parse Oracle DB address list from: {}", connectionUrl);
```

Comment:
do we really want to continue parsing in case of error ?

Reply:
Could be an error or an unknown use of the connection string. However, I only care about very specific nodes in this tree, so if the parsing yields a valid tree in which I can find the tree nodes I am looking for, I don't care.